File: co.yat.android.ExampleView/src/co/yat/android/exampleview/MainActivity.java
package co.yat.android.exampleview;

import android.os.Bundle;
import android.app.Activity;
import android.util.Log;
import android.view.Menu;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

public abstract class MainActivity extends Activity {

	static String APP_NAME = "Main Android Library";
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		Log.v(APP_NAME,"Creating Bundle 1.1");
		ShowView1();
		
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.activity_main, menu);
		return true;
	}
	
	public void GetMessage(String uri, String json) {
		
		Log.v(APP_NAME,"Get Message " + uri);
		
		if(uri.equals("/ExampleViewController/View2/{Show}")) {
			ShowView2();
		}
		
		if(uri.equals("/ExampleViewController/View2/{Hide}")) {
			ShowView1();
		}
		
	}
	
	private void ShowView1() {
		
		setContentView(R.layout.activity_main);
	    // Capture our button from layout
	    Button button = (Button)findViewById(R.id.button1);
	    // Register the onClick listener with the implementation above
	    button.setOnClickListener(new View.OnClickListener() {

	          @Override

	          public void onClick(View v) {

	        	  SendMessage("/ExampleViewController/Button/{Click}","{}");

	          }

	      });
		
	}
	
	private void ShowView2() {
		
		setContentView(R.layout.view_two);
		Button button = (Button)findViewById(R.id.button2);
		button.setOnClickListener(new View.OnClickListener() {

	          @Override

	          public void onClick(View v) {

	        	  SendMessage("/View2/CloseButton/{Click}","{}");

	          }

	      });
		
	}
	
	public abstract void SendMessage(String uri, String json);

}

File: co.yat.android.container/MainActivity.cs
using System;

using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;

namespace co.yat.android.container
{
	[Activity (Label = "co.yat.android.container", MainLauncher = true)]
	public class MainActivity : CO.Yat.Android.Exampleview.MainActivity
	{

		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);

			//GetMessage("test","test");

			#if(CALABASH)
				const string keyguard = "keyguard";
				var mKeyGuardManager = (KeyguardManager)GetSystemService (keyguard);
				var mLock = mKeyGuardManager.NewKeyguardLock (this.GetType ().ToString ());
				mLock.DisableKeyguard ();
			#endif

		}

		public override void SendMessage (String uri, String json)
		{

			Console.WriteLine("Got Message " + uri);

			switch (uri) {
			case "/ExampleViewController/Button/{Click}":
				base.GetMessage("/ExampleViewController/View2/{Show}","{}");
				break;
			case "/View2/CloseButton/{Click}":
				GetMessage("/ExampleViewController/View2/{Hide}","{}");
				break;
			}


		}

	}
}


